### PR TITLE
[WIP] QMAPS-1046 : events list

### PR DESCRIPTION
**Pull request by xem**
## Description
- Create events yml and jsx files inspired by Category
- Perform event type check (return to home if unknown)
- Perform AJAX request to the events API endpoint
- WIP: display list items and markers properly

## Why
To have events in Qwant Maps!

## Screenshots

image

image

### CODE REVIEW

File: src/adapters/poi/idunn_poi.js [hunk 1]
```diff
@@ -87,6 +87,36 @@ export default class IdunnPoi extends Poi {
     }
   }
 
+  static async poiEventLoad(bbox, size, category, query) {
+    const url = `${serviceConfig.idunn.url}/v1/events`;
+    const requestParams = { bbox, size };
+    if (category) {
+      requestParams['category'] = category;
+    }
+    if (query) {
```
Comment (bbecquet) [inline src/adapters/poi/idunn_poi.js]: So is this parameter useful after all?

File: src/adapters/poi/idunn_poi.js [hunk 2]
```diff
@@ -87,6 +87,36 @@ export default class IdunnPoi extends Poi {
...
+      console.log(response);
+      response.events = response.events.map(rawPoi => new IdunnPoi(rawPoi));
+      return response;
+    } catch (err) {
+      if (err === 400 || err === 404 ) {
+        return {};
+      } else {
+        const s_requestParams = JSON.stringify(requestParams);
+        Error.sendOnce(
+          'idunn_poi', 'poiApiLoad',
```
Comment (bbecquet) [inline src/adapters/poi/idunn_poi.js]: `poiApiLoad` => `poiEventLoad`

File: src/panel/event/EventListPanel.jsx [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+    if (panelContent) {
+      panelContent.scrollTop = 0;
+    }
+  }
+
+  componentWillUnmount() {
+    window.unListen(this.mapMoveHandler);
+  }
+
+  fitMapAndFetch() {
```
Comment (bbecquet) [inline src/panel/event/EventListPanel.jsx]: I wonder if this dirty stuff, among other things, can be factorized between categories and events… I really thought events were going to be more distinct than categories, but for now it looks exactly the same :|

Comment (xem) [reply src/panel/event/EventListPanel.jsx]: Yes, I think a lot of things will be in common... basically, everything besides the list header, list items's look and the fact that we keep osm pois or not. And as you mentioned earlier, categories need a "query" param but events don't.

File: src/adapters/poi/idunn_poi.js (hunk 1 above)
Comment (xem) [reply src/adapters/poi/idunn_poi.js]: no, I cleaned it up but forgot to ûsh my last commit due to a surprise meeting :)

File: src/panel/app_panel.js [hunk 4]
```diff
@@ -281,6 +290,19 @@ export default class AppPanel {
     });
   }
 
+  openEvents(params) {
+    const { type: eventName, q: query, ...otherOptions } = params;
```
Comment (bbecquet) [inline src/panel/app_panel.js]: The `q: query` part can be removed here and everywhere else (props especially), as the API doesn't support (or at least it's not a parameter used by `poiEventLoad`).

File: src/panel/app_panel.js [hunk 5]
```diff
@@ -21,24 +22,28 @@ import Router from 'src/proxies/app_router';
 import Poi from 'src/adapters/poi/poi.js';
 import layouts from './layouts.js';
 import ReactPanelWrapper from 'src/panel/reactPanelWrapper';
+import events from '../../config/events.yml';
```
Comment (bbecquet) [inline src/panel/app_panel.js]: Just a note that you can also write `config/events.yml` as there is an alias defined.

File: src/panel/app_panel.js [hunk 6]
```diff
@@ -254,12 +268,7 @@ export default class AppPanel {
     this.activePoiId = null;
     this.panels
       .filter(panel => panel !== panelToOpen)
-      .forEach(panel => {
-        if (panel === this.categoryPanel) {
```
Comment (bbecquet) [inline src/panel/app_panel.js]: If you remove this test, the category markers won't get removed after leaving the category panel. Instead of removing it, I think you need to adapt it for the `eventListPanel` instead.

File: src/panel/event/EventListPanel.jsx [hunk 7]
```diff
@@ -0,0 +1,155 @@
...
+// import PoiCategoryItemList from '../category/PoiCategoryItemList';
+
+const eventConfig = nconf.get().events;
+const MAX_PLACES = Number(eventConfig.maxPlaces);
+
+class EventListPanel extends React.Component {
+
+  static propTypes = {
+    eventName: PropTypes.string,
+    query: PropTypes.string,
```
Comment (bbecquet) [inline src/panel/event/EventListPanel.jsx]: Related to the comment above, `query` is not used so can be removed.

File: src/panel/app_panel.js [hunk 8]
```diff
@@ -254,12 +268,7 @@ export default class AppPanel {
...
-          fire('remove_category_markers');
-        }
-        panel.close();
-      });
+      .forEach(panel => { if (panel.close) { panel.close(); }});
```
Comment (bbecquet) [inline src/panel/app_panel.js]: All panels have a `close` method, I don't think you need this test.

File: src/panel/app_panel.js (hunk 6 above)
Comment (xem) [reply src/panel/app_panel.js]: indeed, I think a bad manip made me remove these lines. Fixed :)

File: src/panel/app_panel.js (hunk 8 above)
Comment (xem) [reply src/panel/app_panel.js]: indeed, I added it when I encountered "no function panel.close" errors, but the problem didn't come from here!

File: src/panel/event/PoiEventItem.jsx [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+import PoiTitleImage from 'src/panel/poi/PoiTitleImage';
+
+const PoiEventItem = ({ poi, eventName }) => {
+  const address = poi.address || {};
+
+  const blocks = poi.blocks;
+  const dates_block = blocks.find(block => block.type === 'event_opening_dates') || {};
+  let date_start, date_end, dates_string;
+
+  if (dates_block) {
```
Comment (bbecquet) [inline src/panel/event/PoiEventItem.jsx]: I thought you wanted to do that in another PR and use the Intl API…
